frontend/model/contracts/chatroom.js

```diff
@@ -129,17 +129,24 @@ function createNotificationData (
 }
 
 function emitMessageEvent ({ type, contractID, hash, state }: {
-  type: string, contractID: string, hash: string, state: Object
+  type: string,
+  contractID: string,
+  hash: string,
+  state?: Object
 }): void {
-  for (let i = state.messages.length - 1; i >= 0; i--) {
-    if (state.messages[i].id === hash) {
-      sbp('okTurtles.events/emit', `${CHATROOM_MESSAGE_ACTION}-${contractID}`, {
-        type,
-        data: { message: state.messages[i] }
-      })
-      break
+  let data = {}
+  if (type === MESSAGE_ACTION_TYPES.ADD_MESSAGE || type === MESSAGE_ACTION_TYPES.EDIT_MESSAGE) {
```

Comment:
If `state` is `null` or `undefined`, then this section of the `if` statement won't do anything.

So instead of having constant checks for `state?.` below, couldn't you add `state` to the condition here? E.g. something like:

```js
  if (state && (type === MESSAGE_ACTION_TYPES.ADD_MESSAGE || type === MESSAGE_ACTION_TYPES.EDIT_MESSAGE)) {
```

BUT - an even better question - why are you making `state` optional here? Isn't it passed in everywhere this function is called? If so, then you can update the function parameters to indicate it mandatory, and remove the `state?` stuff below and just make it `state.messages`

Reply:
Correct. Just make `state` parameter mandatory and remove `?`. Nice catch. Thanks